kedro/config/omegaconf_config.py

```diff
@@ -199,18 +208,19 @@ def __getitem__(self, key) -> dict[str, Any]:
 
         config.update(env_config)
 
-        if not processed_files:
+        if not processed_files and key != "globals":
             raise MissingConfigException(
                 f"No files of YAML or JSON format found in {base_path} or {env_path} matching"
                 f" the glob pattern(s): {[*self.config_patterns[key]]}"
             )
         return config
 
-    def __repr__(self):  # pragma: no cover
-        return (
-            f"OmegaConfigLoader(conf_source={self.conf_source}, env={self.env}, "
-            f"config_patterns={self.config_patterns})"
-        )
+    def _get_conf_path(self, env):
+        if self._protocol == "file":
+            conf_path = str(Path(self.conf_source) / env)
+        else:
+            conf_path = str(Path(self._fs.ls("", detail=False)[-1]) / env)
+        return conf_path
 
     def load_and_merge_dir_config(  # noqa: too-many-arguments
```

Comment:
I think we need to have a closer look at which methods are now "public" and which "private". I know python doesn't strictly enforce these concepts, but initially the `OmegaConfigLoader` was built in a way that all "public" methods could easily be discovered and overwritten by users who want to build their own custom version on top of the class. Now `load_and_merge_dir_config` is still public but called from a private method `_get_config()`. Did you mark methods public/private with a specific reason in mind?

Reply:
My intention was to not change the private or public nature or the logic used by pre-existing functions. 
* `load_and_merge_dir` is still public and just certain parts of it have been taken out as helper methods. I don't think this would affect anyone who has overwritten `load_and_merge_dir()` method in their custom implementations.
    * The refactoring of `load_and_merge_dir()` is not necessarily needed to make globals work tbh, I can revert it but I did see @noklam had proposed something similar in #2591 
* `__getitem__()` is a magic function which is treated essentially like a private method (it's not meant to be called directly) so I would think it doesn't makes much of a difference if some of it's code is separated into another private method. It can still be overwritten by the users as well. 
    * This change though is required for this implementation to work